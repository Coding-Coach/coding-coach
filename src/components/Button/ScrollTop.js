import React from 'react';

import ArrowUp from '../Icon/ArrowUp';
import Button from './index';

const styles = {
  hide: {
    bottom: 20,
    right: -65,
  },
  show: {
    right: 20,
  },
};

export default class ScrollTop extends React.Component {
  state = {
    scrollTopClass: '',
  };

  componentDidMount() {
    this.scrollHandler();
    window.addEventListener('scroll', this.scrollHandler);
  }

  componentWillUnmount() {
    window.removeEventListener('scroll', this.scrollHandler);
  }

  scrollHandler = () => {
    const scrollToTopClass = window.pageYOffset > window.innerHeight / 4 ? 'opacity-100' : '';

    if (scrollToTopClass !== this.state.scrollToTopClass) {
      this.setState({
        scrollToTopClass,
      });
    }
  };

  scrollToTop = () => {
    window.scrollTo({
      behavior: 'smooth',
      top: 0,
    });
  };

  render() {
    const { scrollToTopClass } = this.state;

    return (
      <div
        aria-label="Scroll to top"
        tabIndex={scrollToTopClass ? 0 : -1}
        className={`transition-slow fixed cursor-pointer opacity-0 z-10 ${scrollToTopClass}`}
        style={{
          ...styles.hide,
          ...(scrollToTopClass && styles.show),
        }}
      >
        <Button onClick={this.scrollToTop} onKeyPress={this.scrollToTop} className="px-3">
          <ArrowUp width={27} height={28} color="currentColor" />
        </Button>
      </div>
    );
  }
}
