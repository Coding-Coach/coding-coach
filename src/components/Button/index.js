import React from 'react';
import { any, func, string } from 'prop-types';
import classNames from 'classnames';
import { Link } from 'gatsby';

const Button = React.memo(({ children, className, to, onClick }) => {
  const css = classNames('inline-block bg-primary text-white px-6 py-3 rounded hover:bg-primary-dark', className);

  if (to) {
    if (to.startsWith('http')) {
      return (
        <a href={to} className={css}>{children}</a>
      );
    }

    return (
      <Link to={to} className={css}>{children}</Link>
    );
  }

  return (
    <button onClick={onClick} className={css}>{children}</button>
  );
});

Button.propTypes = {
  children: any,
  className: string,
  onClick: func,
};

export default Button;
