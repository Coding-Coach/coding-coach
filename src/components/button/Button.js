import React from 'react';
import styles from './button.scss';
import classNames from 'classnames';
import PropTypes from 'prop-types';
/**
 *
 * @param {children} props
 * @param {type} props  primary, danger, secondary
 * @param {size} props  default, large, small,
 * @param {disabled} props  true, false
 * @param {fullWidth} props Button width is 100% of parent
 */
const Button = (props) => {
  const { children, type, size, disabled, fullWidth, id, onClick } = props;
  const classes = classNames(
    styles.button,
    styles[type] || styles.primary,
    styles[size] || styles.default,
    fullWidth && styles.fullWidth,
  );
  return (
    <button id={id} disabled={disabled} onClick={onClick} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  size: PropTypes.string,
  type: PropTypes.string,
  children: PropTypes.node.isRequired,
  disabled: PropTypes.bool,
  fullWidth: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  id: PropTypes.string,
};

export default Button;
