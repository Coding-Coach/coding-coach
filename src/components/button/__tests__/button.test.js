import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import Button from './../Button';

test('renders the button component', () => {
  const { container } = render(<Button onClick={() => {}}>CLICK ME</Button>);
  expect(container).toMatchSnapshot();
});

test('Handle Click', () => {
  const handleClick = jest.fn();
  const { getByText } = render(<Button onClick={handleClick}>SUBMIT</Button>);
  fireEvent.click(getByText('SUBMIT'));
  expect(handleClick).toHaveBeenCalledTimes(1);
});
