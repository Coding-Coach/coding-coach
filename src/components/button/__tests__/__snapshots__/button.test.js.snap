// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the button component 1`] = `
<div>
  <button
    class="button primary medium rounded uppercase"
  >
    CLICK ME
  </button>
</div>
`;
